server/api/users.js

```diff
@@ -1054,23 +1056,37 @@ router.put('/users/:id/civic/trial', jwtAuth('write'), async (req, res, next) =>
       return;
     }
 
-    const doc = await dbRef.doc(id).get();
-    if (!doc.exists) throw new Error('USER_NOT_EXIST');
+    const userDoc = await dbRef.doc(id).get();
+    if (!userDoc.exists) throw new Error('USER_NOT_EXIST');
+
+    const configDoc = await configRef.doc('civicLiker').get();
+    const preRegQuota = (configDoc.exists && configDoc.data().preRegQuota) || 0;
 
-    const updateObj = {
-      isPreRegCivicLiker: true,
-    };
+    const preRegQuery = await dbRef
+      .where('isPreRegCivicLiker', '==', true)
+      .limit(preRegQuota)
+      .get();
+    const preRegCount = preRegQuery.docs.length;
+
+    const updateObj = {};
+    if (preRegCount >= preRegQuota) {
+      updateObj.isOnPreRegCivicLikerWaitingList = true;
```

Comment:
use a string field that act as a enum i.e.`PreRegCivicLikerStatus = 'waiting_list'` in case we want to do more judgement later?